pkg/registry/file/vulnsummarystorage_test.go

```diff
@@ -0,0 +1,461 @@
+			key:  "/spdx.softwarecomposition.kubescape.io",
+			want: 4,
+		},
+		{
+			name: "from top",
+			key:  "/",
+			want: 5,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+			_ = fs.Mkdir(DefaultStorageRoot, 0755)
+			for _, f := range files {
+				_ = afero.WriteFile(fs, DefaultStorageRoot+f, []byte(""), 0644)
+			}
+			s := NewStorageImpl(fs, DefaultStorageRoot)
+			got, err := s.Count(tt.key)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Count() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("Count() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestVulnSummaryStorageImpl_Create(t *testing.T) {
+	type args struct {
+		key string
+		obj runtime.Object
+		out runtime.Object
+		in4 uint64
+	}
+	tests := []struct {
+		name     string
+		readonly bool
+		args     args
+		wantErr  bool
+		want     runtime.Object
+	}{
+		{
+			name: "not supported",
+			args: args{
+				key: "/spdx.softwarecomposition.kubescape.io/vulnerabilitysummaries/kubescape/toto",
+				obj: &v1beta1.VulnerabilitySummary{},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fs := afero.NewReadOnlyFs(afero.NewMemMapFs())
+			s := NewVulnSummaryStorageImpl(fs, DefaultStorageRoot)
+			err := s.Create(context.TODO(), tt.args.key, tt.args.obj, tt.args.out, tt.args.in4)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+		})
+	}
+}
+
+func TestVulnSummaryStorageImpl_Delete(t *testing.T) {
+	type args struct {
+		key                  string
+		obj                  runtime.Object
+		precondition         *storage.Preconditions
+		validateDeletionFunc storage.ValidateObjectFunc
+		cachedObj            runtime.Object
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "not supported",
+			args: args{
+				key: "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape/toto",
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fs := afero.NewReadOnlyFs(afero.NewMemMapFs())
+			s := NewVulnSummaryStorageImpl(fs, DefaultStorageRoot)
+			err := s.Delete(context.TODO(), tt.args.key, tt.args.obj, tt.args.precondition, tt.args.validateDeletionFunc, tt.args.cachedObj)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+		})
+	}
+}
+
+func Test_getVulnManifestSummaryDirPath(t *testing.T) {
+
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{
+			input:    "/spdx.softwarecomposition.kubescape.io/vulnerabilitysummaries/default/default",
+			expected: "/data/spdx.softwarecomposition.kubescape.io/vulnerabilitymanifestsummaries",
+		},
+	}
+
+	s := VulnSummaryStorageImpl{
+		appFs: afero.NewMemMapFs(),
+		root:  "/data",
+	}
+
+	for _, td := range tests {
+		res := s.getVulnManifestSummaryDirPath(td.input)
+		assert.Equal(t, td.expected, res)
+	}
+}
```

Comment:
issue: after the architectural changes requested above, this should not be tested